ocs_ci/ocs/cluster.py

```diff
@@ -644,6 +644,47 @@ def get_rebalance_status(self):
                 and states['count'] == total_pg_count
             )
 
+    def wait_for_rebalance(self, timeout=None):
+        """
+        Wait for re-balance to complete
+
+        Args:
+            timeout (int): Time to wait for the completion of re-balance.
+                In case IO is running in the test background (according to --io-in-bg
+                run-ci param, the timeout will be determined by the amount of load
+                (io_load run-ci param)
+
+        Returns:
+            bool: True if rebalance completed, False otherwise
+
+        """
+        if config.RUN.get('io_in_bg'):
+            io_load = config.RUN.get('io_load')
+            if 10 < io_load <= 30:
+                timeout = 900
+            if 30 < io_load <= 50:
+                timeout = 1800
+            if 50 < io_load <= 70:
+                timeout = 2700
+            if io_load > 70:
+                timeout = 3600
```

Comment:
With heavy load I think that even 1 hour is not enough.  But let's have some testing.  It maybe also depends on number of OSDs we have and amount of data.  Maybe worth to ask some Ceph guys if we can somehow calculate this more specific.

Reply:
Asked here:  https://bugzilla.redhat.com/show_bug.cgi?id=1856975#c9. So hope we will get some answer, maybe from Josh.

Reply:
the suggested solution from Josh is to implement something similar to https://github.com/ceph/ceph/blob/master/qa/tasks/ceph_manager.py#L2576.

See http://post-office.corp.redhat.com/archives/rhocs-eng/2020-July/msg00325.html

Reply:
For now, in order to unblock the upgrade executions, I will set a static timeout of 1800 for add capacity and add node tests.
We can think about implementing a solution as suggested above, but this will take some time to get merged

Reply:
changed to 3600 seconds timeout